packages/jazz-tools/src/coValues/coList.ts

```diff
@@ -446,6 +449,77 @@ export class CoList<Item = any> extends Array<Item> implements CoValue {
     ): InstanceType<Cl> {
         return cl.fromRaw(this._raw) as InstanceType<Cl>;
     }
+
+    asPlainData<L extends CoList>(this: L): RecursiveCoMapInit<L>;
+    asPlainData<L extends CoList, Depth>(
+        this: L,
+        depth: Depth & DepthsIn<L>,
+    ): Promise<RecursiveCoMapInit<L> | undefined>;
+    asPlainData<L extends CoList, Depth>(
+        this: L,
+        depthOrSeen?:
```

Comment:
this signature is a bit unwieldy and we don't really want to expose the "seen" logic to the user, even in the type. Please may you pull out a helper function that does the recursion, with separate parameters for depth and seen and then we can keep the signature here clean? Thanks!